SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorLogrotateAdditionalFile.java

```diff
@@ -0,0 +1,36 @@
+package com.hubspot.singularity.executor.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+public class SingularityExecutorLogrotateAdditionalFile {
+    private final String filename;
+    private final Optional<String> extension;
+    private final Optional<String> dateformat;
+
+    public static SingularityExecutorLogrotateAdditionalFile fromString(String value) {
```

Comment:
Dang -- I forgot the annotation but Jackson is magically making it work (based on the method name & signature i assume). I'll add the annotation.